code/datums/elements/dirty_glasses.dm

```diff
@@ -0,0 +1,62 @@
+/**
+  * Can be applied to glasses to cause them to apply blurry vision and stuff
+  */
+/datum/element/dirty_glasses
+
+/datum/element/dirty_glasses/Attach(datum/target)
+	. = ..()
+	if(!istype(target,/obj/item/clothing/glasses))
+		return ELEMENT_INCOMPATIBLE
+
+	RegisterSignal(target, list(COMSIG_ITEM_EQUIPPED, COMSIG_ITEM_DROPPED), .proc/equippedChanged)
+
+	RegisterSignal(target,COMSIG_COMPONENT_CLEAN_ACT,.proc/clean)
+
+	RegisterSignal(target,COMSIG_PARENT_EXAMINE,.proc/examine)
+
+	var/obj/item/clothing/glasses/glassies = target
+	if(!ishuman(glassies.loc))
```

Comment:
shouldn't this be `ismob`?

Reply:
only humans can have glasses

Reply:
then why are you use mob?

Reply:
because all the slots are on /mob, but only /human can only use this, there is no reason to upscale it to human, since im not accessing any functions or variables that are declared there.